File: android/src/main/kotlin/com/innovationcodes/tuya_home_sdk_flutter/DeviceDiscoveryStreamHandler.kt
package com.innovationcodes.tuya_home_sdk_flutter

import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.EventChannel.EventSink


class DeviceDiscoveryStreamHandler : EventChannel.StreamHandler {
    var discoverySink: EventSink? = null
    override fun onListen(arguments: Any?, events: EventSink?) {
        this.discoverySink = events
    }

    override fun onCancel(arguments: Any?) {
        this.discoverySink = null
    }
}

File: ios/Classes/DeviceDiscoveryStreamHandler.swift
//
//  DeviceDiscoveryStreamHandler.swift
//  Pods
//
//  Created by Basem Abduallah on 06/08/2025.
//
import Flutter

class DeviceDiscoveryStreamHandler: NSObject, FlutterStreamHandler{
    var discoverySink: FlutterEventSink?

    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.discoverySink = events
        
        return nil
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.discoverySink = nil
        
        return nil
    }

   
}
